### PR TITLE
[Telemetry Operator] Make ValueFrom and SecretRef nullable

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

Changes proposed in this pull request:

- Rename `ValueFromType` to `ValueFromSource` (similar to https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2075)
- Make `ValueFrom` field nullable. If `Value` is defined, it must be set to null
- Make `SecretRef` optional (in the future we might want to add support for config maps) 

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/15134

### CODE REVIEW

File: components/telemetry-operator/internal/utils/envvar/format.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
 )
 
-func GenerateName(logpipelineName string, secretKeyRef v1alpha1.SecretKeyRef) string {
+func GenerateName(logpipelineName string, secretKeyRef v1alpha1.SecretRef) string {
```
Comment (lindnerby) [inline components/telemetry-operator/internal/utils/envvar/format.go:10]: Rename input params as well please

File: components/telemetry-operator/apis/telemetry/v1alpha1/logpipeline_types.go [hunk 2]
```diff
@@ -140,28 +140,32 @@ type FileMount struct {
...
 
-func (v *ValueFromType) IsSecretRef() bool {
-	return v.SecretKey.Name != "" && v.SecretKey.Key != ""
+type ValueFromSource struct {
+	SecretRef *SecretRef `json:"secretKeyRef,omitempty"`
 }
 
-type ValueFromType struct {
-	SecretKey SecretKeyRef `json:"secretKeyRef,omitempty"`
+func (v *ValueFromSource) IsSecretRef() bool {
```
Comment (lindnerby) [inline components/telemetry-operator/apis/telemetry/v1alpha1/logpipeline_types.go:164]: Maybe this should be on the ValueType? Why ask the non-nil SecretRef if it "IsSecretRef"? Or rename it to something else if it's just for validation?

**Review by lindnerby — CHANGES_REQUESTED** [review]

File: components/telemetry-operator/internal/utils/envvar/format.go (hunk 1 above)
Comment (skhalash) [reply components/telemetry-operator/internal/utils/envvar/format.go]: Reverted back to `SecretKeyRef`

File: components/telemetry-operator/apis/telemetry/v1alpha1/logpipeline_types.go (hunk 2 above)
Comment (skhalash) [reply components/telemetry-operator/apis/telemetry/v1alpha1/logpipeline_types.go]: After a discussion, we decided to keep it like this so far

**Review by lindnerby — APPROVED** [review]

**Pull request merged**